src/overlays/actors/ovl_En_Kaizoku/z_en_kaizoku.c

```diff
@@ -63,120 +133,1983 @@ ActorInit En_Kaizoku_InitVars = {
+    this->exitIndex = KAIZOKU_GET_EXIT_INDEX(&this->picto.actor);
+    this->switchFlag = KAIZOKU_GET_SWITCHFLAG(&this->picto.actor);
+
+    if (this->unk_2CA >= 2) {
+        this->unk_2CA = 0;
+    } else if (this->unk_2CA == 2) {
+        this->unk_2CA = 0;
+    }
+
+    this->unk_2EC = this->picto.actor.world.rot.z;
+    this->picto.actor.world.rot.z = 0;
+    this->picto.actor.colChkInfo.damageTable = &sDamageTable;
+    SkelAnime_InitFlex(play, &this->skelAnime, &gKaizokuSkel, &object_kz_Anim_0058B8, this->jointTable,
+                       this->morphTable, KAIZOKU_LIMB_MAX);
+    Collider_InitAndSetCylinder(play, &this->unk_3D4, &this->picto.actor, &sCylinderInit);
+    Collider_InitAndSetQuad(play, &this->unk_420, &this->picto.actor, &sQuadInit);
+    blureInit.p1StartColor[0] = blureInit.p1StartColor[1] = blureInit.p1StartColor[2] = blureInit.p1StartColor[3] =
+        blureInit.p2StartColor[0] = blureInit.p2StartColor[1] = blureInit.p2StartColor[2] = blureInit.p1EndColor[0] =
+            blureInit.p1EndColor[1] = blureInit.p1EndColor[2] = blureInit.p2EndColor[0] = blureInit.p2EndColor[1] =
+                blureInit.p2EndColor[2] = 255;
+    blureInit.p2StartColor[3] = 64;
+    blureInit.p1EndColor[3] = blureInit.p2EndColor[3] = 0;
+    blureInit.elemDuration = 8;
+    blureInit.unkFlag = 0;
+    blureInit.calcMode = 2;
+    Effect_Add(play, &this->unk_3D0, EFFECT_BLURE1, 0, 0, &blureInit);
+    Actor_SetScale(&this->picto.actor, 0.0125f);
+    this->picto.actor.flags |= ACTOR_FLAG_8000000;
+    this->picto.actor.flags &= ~ACTOR_FLAG_1;
+    if (this->switchFlag == 127) {
+        this->switchFlag = -1;
+    }
+
+    if ((this->switchFlag >= 0) && Flags_GetSwitch(play, this->switchFlag)) {
+        Actor_Kill(&this->picto.actor);
+        return;
+    }
+
+    this->unk_3D4.dim.radius = 20;
+    this->unk_3D4.dim.height = 65;
+    this->unk_3D4.dim.yShift = 0;
+    this->unk_2D6 = this->picto.actor.cutscene;
+    this->picto.actor.world.pos.y = player->actor.world.pos.y + 160.0f;
+    this->picto.validationFunc = EnKaizoku_ValidatePictograph;
+    this->picto.actor.flags |= ACTOR_FLAG_400;
+    func_80B85F48(this);
+}
+
+void EnKaizoku_Destroy(Actor* thisx, PlayState* play) {
+    EnKaizoku* this = THIS;
+
+    Effect_Destroy(play, this->unk_3D0);
+    Collider_DestroyCylinder(play, &this->unk_3D4);
+    Collider_DestroyQuad(play, &this->unk_420);
+    Audio_RestorePrevBgm();
+}
+
+s32 func_80B85858(EnKaizoku* this, PlayState* play) {
+    Actor* itemAction;
+    Player* player = GET_PLAYER(play);
+    s16 sp1E = this->unk_2B0;
+
+    for (itemAction = play->actorCtx.actorLists[ACTORCAT_ITEMACTION].first; itemAction != NULL;
+         itemAction = itemAction->next) {
+        //! FAKE:
+        if (1) {}
+        if ((itemAction->id == ACTOR_ARMS_HOOK || itemAction->id == ACTOR_EN_ARROW) && player->unk_D57) {
+            if ((itemAction->id != ACTOR_ARMS_HOOK || this->unk_2B0 != 13) && this->unk_2B0 != 6) {
```

Comment:
```suggestion
        if (((itemAction->id == ACTOR_ARMS_HOOK) || (itemAction->id == ACTOR_EN_ARROW)) && player->unk_D57) {
            if (((itemAction->id != ACTOR_ARMS_HOOK) || (this->unk_2B0 != 13)) && (this->unk_2B0 != 6)) {
```
etc., applies elsewhere. Also `unk_D57` doesn't seem to be boolean?